### PR TITLE
[WIP] compute next version from ../dmd/VERSION

### PR DESCRIPTION
**Pull request by MartinNowak**
- allows to build changelog preview during beta

### CODE REVIEW

**Comment by MartinNowak** [comment]
As the changelog folders are currently only purged when the release is out, the changelog preview on master will contain stuff that's going to be released earlier.
To fix that, we'd want to update the changed.d tool to support incremental updates so that we can purge changelogs right away, see https://trello.com/c/Lu1mDNYO/336-rework-changelog-generation-at-dlangorg?menu=filter&filter=@dawg.

File: posix.mak [hunk 1]
```diff
@@ -63,7 +75,6 @@ PHOBOS_LATEST_DIR_GENERATED=$(GENERATED)/phobos-release
 #   Makefile dependencies and rules
 PHOBOS_FILES := $(shell find $(PHOBOS_DIR) -name '*.d' -o -name '*.mak' -o -name '*.ddoc')
 PHOBOS_FILES_GENERATED := $(subst $(PHOBOS_DIR), $(PHOBOS_DIR_GENERATED), $(PHOBOS_FILES))
-$(shell [ ! -d $(PHOBOS_DIR) ] && git clone --depth=1 ${GIT_HOME}/phobos $(PHOBOS_DIR))
 $(shell [ ! -d $(PHOBOS_LATEST_DIR) ] && git clone -b v${LATEST} --depth=1 ${GIT_HOME}/phobos $(PHOBOS_LATEST_DIR))
```
Comment (wilzbach) [inline posix.mak]: I didn't forget this line - `PHOBOS_LATEST_DIR` is defined later (L45).

**Comment by MartinNowak** [comment]
Think, we should do this with proper targets and dependencies, instead of unconditional execution.
It's really annoying that the clean target clones repos.

**Comment by wilzbach** [comment]
(rebased to merge, merge-conflicts were trivial and only in `changelog/index.dd`)

**Comment by wilzbach** [comment]
Okay as I want to see this moving forward, I used the core bit of this PR and put it into another PR (https://github.com/dlang/dlang.org/pull/1944/), which should be a lot easier to merge. Afterward, moving forward with this PR, should become easier as well.

**Pull request closed without merging**